gtsfm/densify/mvs_patchmatchnet.py

```diff
@@ -0,0 +1,280 @@
+"""Inference methods for GtsfmData in PatchmatchNet
+
+Authors: Ren Liu
+"""
+import time
+from typing import Dict, List
+
+import numpy as np
+import torch
+import torch.nn as nn
+from torch.utils.data import DataLoader
+
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.common.image import Image
+from gtsfm.densify.mvs_base import MVSBase
+import gtsfm.densify.mvs_utils as mvs_utils
+from gtsfm.densify.patchmatchnet_data import PatchmatchNetData
+from gtsfm.utils import logger as logger_utils
+from thirdparty.patchmatchnet.models.net import PatchmatchNet
+import thirdparty.patchmatchnet.utils as patchmatchnet_utils
+import thirdparty.patchmatchnet.eval as patchmatchnet_eval_funcs
+
+logger = logger_utils.get_logger()
+
+torch.backends.cudnn.benchmark = True
+PATCHMATCHNET_WEIGHTS_PATH = "thirdparty/patchmatchnet/checkpoints/model_000007.ckpt"
+
+# all default values are assigned by Wang et al. https://github.com/FangjinhuaWang/PatchmatchNet/blob/main/eval.py
+DEFAULT_BATCH_SIZE = 1
+DEFAULT_VIEW_NUMBER = 5
+DEFAULT_WORKERS_NUMBER = 4
+
+DEFAULT_GEOMETRIC_PIXEL_THRESH = 1.0
+DEFAULT_GEOMETRIC_DEPTH_THRESH = 0.01
+DEFAULT_PHOTOMETRIC_THRESH = 0.8
+
+DEFAULT_INTERVAL_SCALE = [0.005, 0.0125, 0.025]
+DEFAULT_PROPAGATION_RANGE = [6, 4, 2]
+DEFAULT_ITERATION_NUMBER = [1, 2, 2]
+DEFAULT_SAMPLE_NUMBER = [8, 8, 16]
+DEFAULT_PROPAGATE_NEIGHBORS = [0, 8, 16]
+DEFAULT_EVALUATE_NEIGHBORS = [9, 9, 9]
+
+DEFAULT_MINIMUM_CONSISTENT_VIEW_NUMBER = 3
```

Comment:
I prefer `DEFAULT_MIN_NUM_CONSISTENT_VIEWS`. Can you add a 1-liner inline comment explaining what this means, too?